core/src/test/java/com/linecorp/armeria/server/AnnotatedHttpServiceTest.java

```diff
@@ -315,13 +319,76 @@ public String glob2(ServiceRequestContext ctx) {
 
         @Get
         @Path("regex:^/regex/(?<path>.*)$")
-        public String regex(ServiceRequestContext ctx, @PathParam("path") String path) {
+        public String regex(ServiceRequestContext ctx, @Param("path") String path) {
             return "regex:" + ctx.path() + ':' + path;
         }
     }
 
+    @Converter(target = String.class, value = UnformattedStringConverter.class)
+    public static class MyAnnotatedService7 {
+        @Get
+        @Path("/param/get")
+        public String paramGet(RequestContext ctx,
+                               @Param("username") String username,
+                               @Param("password") String password) {
+            validateContext(ctx);
+            return username + "/" + password;
+        }
+
+        @Post
+        @Path("/param/post")
+        public String paramPost(RequestContext ctx,
+                                @Param("username") String username,
+                                @Param("password") String password) {
+            validateContext(ctx);
+            return username + "/" + password;
+        }
+
+        @Get
+        @Path("/map/get")
+        public String mapGet(RequestContext ctx, ParamMap parameters) {
+            validateContext(ctx);
+            return parameters.getFirst("username") + "/" + parameters.getFirst("password");
+        }
+
+        @Post
+        @Path("/map/post")
+        public String mapPost(RequestContext ctx, ParamMap parameters) {
+            validateContext(ctx);
+            return parameters.getFirst("username") + "/" + parameters.getFirst("password");
+        }
+
+        @Get
+        @Path("/param/default1")
+        public String paramDefault1(RequestContext ctx,
+                               @Param(value = "username", defaultValue = "hello") String username,
+                               @Param(value = "password", required = false, defaultValue = "world")
```

Comment:
I wonder if the following would be any better:

```java
public String paramDefault1(
        RequestContext ctx, 
        @Param("username") String username, // Required with no default
        @Param("password") @NotRequired("world") String password, // Optional with a default value
        @Param("captcha") @NotRequired String captcha, // Optional with no default
        ...) { ... }
```

Just an idea. I think it may be better because a user doesn't have to add the ugly `value = ` when adding more attributes. What do you think?